rmf_fleet_adapter/src/rmf_fleet_adapter/events/internal_ReservationNodeNegotiator.hpp

```diff
@@ -0,0 +1,382 @@
+          {
+            RCLCPP_INFO(
+              self->_context->node()->get_logger(),
+              "Reservation: Robot %s is going to final destination %lu",
+              self->_context->name().c_str(),
+              self->_goals[self->_final_allocated_destination.value()->
+              chosen_alternative].waypoint());
+            self->_current_reservation_state = ReservationState::ReceivedResponseProceedImmediate;
+            self->_selected_final_destination_cb(self->_goals[self->
+            _final_allocated_destination.value()->
+            chosen_alternative].waypoint());
+          }
+
+          if (msg->instruction_type
+          == rmf_reservation_msgs::msg::ReservationAllocation::WAIT_IDENTIFIED)
+          {
+            self->_current_reservation_state = ReservationState::ReceivedResponseProceedWaitPoint;
+            self->_selected_waitpoint_cb(self->_waitpoints[self->
+            _final_allocated_destination.value()->
+            chosen_alternative]);
+            RCLCPP_INFO(
+              self->_context->node()->get_logger(),
+              "Reservation: Robot %s is being asked to proceed to a waitpoint %lu",
+              self->_context->name().c_str(),
+              self->_waitpoints[self->_final_allocated_destination.value()->
+              chosen_alternative].waypoint());
+          }
+        });
+
+    context->worker().schedule(
+      [ptr = negotiator->weak_from_this(), w = context->weak_from_this(), same_map](
+        const auto&)
+    {
+      auto context = w.lock();
+      if (!context)
+        return;
+
+      auto negotiator = ptr.lock();
+
+      auto current_location = context->location();
+      if (current_location.size() == 0)
+      {
+        using namespace std::literals::chrono_literals;
+        negotiator->_retry_timer = context->node()->create_wall_timer(
+          500ms, [self = negotiator->weak_from_this(), same_map]()
+          {
+            auto negotiator = self.lock();
+            if(!negotiator)
+            {
+              return;
+            }
+
+            auto current_location = negotiator->_context->location();
+            if (current_location.size() != 0)
+            {
+              negotiator->_retry_timer->cancel();
+            }
+            else
+            {
+              return;
+            }
+            for (std::size_t i = 0; i < negotiator->_goals.size(); ++i)
+            {
+              if (events::wp_name(*negotiator->_context.get(), negotiator->_goals[i]) == negotiator->_context->_get_reserved_location())
+              {
+                RCLCPP_ERROR(negotiator->_context->node()->get_logger(),
+                "Already at goal no need to engage reservation system\n");
+                negotiator->_selected_final_destination_cb(negotiator->_goals[i].waypoint());
+                return;
+              }
+            }
+            RCLCPP_INFO(negotiator->_context->node()->get_logger(),
+            "Sending reservation request");
```

Comment:
Let's include the robot name and the location that's being reserved in order to make this log message useful.